test/parallel/test-https-immutable-options.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+const common = require('../common');
+
+const assert = require('assert');
+const https = require('https');
+const tls = require('tls');
+
+const dftProtocol = {};
+tls.convertNPNProtocols([ 'http/1.1', 'http/1.0' ], dftProtocol);
+
+const opts = { foo: 'bar' };
+const server1 = https.createServer(opts);
+
+assert.deepStrictEqual(opts, { foo: 'bar' });
+assert.strictEqual(server1.NPNProtocols.compare(dftProtocol.NPNProtocols), 0);
+
+const mustNotCall = common.mustNotCall('dummy callback');
```

Comment:
It might be best to just omit the argument here, since if the function is called for some odd reason the default message should be slightly more descriptive and good enough.